DQM/BeamMonitor/plugins/Vx3DHLTAnalyzer.cc

```diff
@@ -1152,6 +1192,88 @@ void Vx3DHLTAnalyzer::endLuminosityBlock (const LuminosityBlock& lumiBlock, cons
 
       delete myLinFit;
       vals.clear();
+
+      // Gaussian fit to 1D vertex coordinate distributions
+      TF1* myGaussFit = new TF1("myGaussFit", "[0]*exp(-(x-[1])*(x-[1])/(2*[2]*[2]))", Vx_Z_Fit->getTH1()->GetXaxis()->GetXmin(), Vx_Z_Fit->getTH1()->GetXaxis()->GetXmax());
+      myGaussFit->SetLineColor(2);
+      myGaussFit->SetLineWidth(2);
+      myGaussFit->SetParName(0,"Ampl.");
+      myGaussFit->SetParName(1,"#mu");
+      myGaussFit->SetParName(2,"#sigma");
+
+      myGaussFit->SetParameter(0, Vx_X_Fit->getTH1()->GetMaximum());
+      myGaussFit->SetParameter(1, Vx_X_Fit->getTH1()->GetMean());
+      myGaussFit->SetParameter(2, Vx_X_Fit->getTH1()->GetRMS());
+      minXfit = Vx_X_Fit->getTH1()->GetBinLowEdge(1);
+      for (int i = 0; i < Vx_X_Fit->getTH1()->GetNbinsX(); i++)
```

Comment:
hi @dinardo - all the loops should be indexed by unsigned ints.

otherwise, have you handled the case of histograms with 0 entries?